agents/hooks/addCurrentAgent.js

```diff
@@ -0,0 +1,19 @@
+const merge = require('ramda/src/merge')
+const isNil = require('ramda/src/isNil')
+
+function addCurrentAgent(hook) {
+  if (isNil(hook.params.credential.agentId)) return hook
```

Comment:
sweet! we maybe need to check if `hook.params.credential` is nil first? could use ramda `path` for this.

Reply:
Nice. Learning a lot about what ramda can do